File: Serene/Serene.Script/Imports/ServerImports/Administration.RoleForm.cs
using jQueryApi;
using Serenity;
using Serenity.ComponentModel;
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace Serene.Administration
{
    public partial class RoleForm : PrefixedContext
    {
        [InlineConstant] public const string FormKey = "Administration.Role";

        public RoleForm(string idPrefix) : base(idPrefix) {}

        public StringEditor RoleName { [InlineCode("{this}.w('RoleName', Serenity.CodeGeneration.ExternalType)")] get; private set; }
    }
}

File: Serene/Serene.Web/Imports/ServerTypings/Administration.RoleForm.ts
namespace Serene.Administration {
    export interface RoleForm {
        RoleName: Serenity.StringEditor;
    }

    export class RoleForm extends Serenity.PrefixedContext {
        static formKey = 'Administration.Role';
        private static init: boolean;

        constructor(prefix: string) {
            super(prefix);

            if (!RoleForm.init)  {
                RoleForm.init = true;

                var s = Serenity;
                var w0 = s.StringEditor;

                Q.initFormType(RoleForm, [
                    'RoleName', w0
                ]);
            }
        }
    }
}

